src/registry/mod.rs

```diff
@@ -0,0 +1,73 @@
+pub mod app;
+pub mod package;
+
+use wasm_bindgen::{prelude::wasm_bindgen, JsValue};
+use wasmer_api::WasmerClient;
+
+use crate::{utils::Error, Wasmer};
+
+static WASMER_CLIENT: std::sync::OnceLock<WasmerClient> = std::sync::OnceLock::new();
+
+#[wasm_bindgen(getter_with_clone)]
+pub struct RegistryConfig {
+    pub registry_url: String,
```

Comment:
This should be an option

Reply:
I think it's good like that, and the [top level of the library](https://github.com/wasmerio/wasmer-js/pull/424/files#diff-372e5b2bec20313bbe06e3457a7a0d632db8ce2633b5e2036911df81fcc1e00aR13) should deal with the correct resolution from the users' input (token as a string -> default registry, otherwise a `RegistryConfig` object). This way, users also have the correct types. Otherwise, to provide the token only but not registry, users would have to use `{registry_url: null, token:"some_token"}` to make linters happy.

Reply:
That's not right. `null` and `undefined` are semantically similar in JS `null == undefined` (this is true).

Basically, you can simply do:

```js
Wasmer.init({token:"some_token"})
// Which is the same as:
Wasmer.init({registry_url: undefined, token:"some_token"})
// Which is "equivalent" to:
Wasmer.init({registry_url: null, token:"some_token"})
```

Reply:
As we want the first one `Wasmer.init({token:"some_token"})` to work, this is the way to make it happen